src/main/java/com/hotels/service/tracing/zipkintohaystack/forwarders/haystack/HaystackDomainConverter.java

```diff
@@ -48,12 +49,22 @@ public static Span fromZipkinV2(zipkin2.Span zipkin) {
         doIfNotNull(zipkin.parentId(), builder::setParentSpanId);
         doIfNotNull(zipkin.localServiceName(), builder::setServiceName);
 
-        if (!isEmpty(zipkin.tags())) {
-            zipkin.tags().forEach((key, value) -> {
-                List<Tag> tagStream = fromZipkinTag(key, value);
-                builder.addAllTags(tagStream);
-            });
-        }
+        List<Tag> tags = zipkin.tags().entrySet().stream()
+                .flatMap(entry -> fromZipkinTag(entry.getKey(), entry.getValue()).stream())
+                .collect(toList());
+        builder.addAllTags(tags);
+
+        var logs = zipkin.annotations().stream()
+                .map(annotation -> Log.newBuilder()
+                        .setTimestamp(annotation.timestamp())
+                        .addFields(Tag.newBuilder()
+                                .setVStr(annotation.value())
+                                .setType(Tag.TagType.STRING)
+                                .setKey("event")
```

Comment:
waiting on the haystack folks to confirm if "event" is a good name for the key.
this has impact on the way data is indexed so we may want to leave this null or configurable.

Reply:
In theory "logs" are not indexed as they are micro-events inside the span so any key should be fine but agree on letting the Haystack folks confirm

Reply:
another way is to use the value as a key and empty string as value.. depends on how the backend works and you'll know best